src/Runtime/ExecutionSession.cpp

```diff
@@ -96,10 +97,27 @@ OMTensorList *ExecutionSession::run(OMTensorList *input) {
   return _entryPointFunc(input);
 }
 
-std::string ExecutionSession::inputSignature() { return _inputSignatureFunc(); }
+void ExecutionSession::setEntryPoint(const std::string &entryPointName) {
+  _entryPointFunc = reinterpret_cast<entryPointFuncType>(
+      _sharedLibraryHandle.getAddressOfSymbol(entryPointName.c_str()));
+  if (!_entryPointFunc) {
+    std::stringstream errStr;
+    errStr << "Cannot load symbol: '" << entryPointName << "'" << std::endl;
+    throw std::runtime_error(errStr.str());
+  }
+  _entryPointName = entryPointName;
+}
+
+const std::string *ExecutionSession::queryEntryPoints() const {
+  return (const std::string *)_queryEntryPointsFunc();
+}
+
+const std::string ExecutionSession::inputSignature() const {
+  return _inputSignatureFunc(_entryPointName.c_str());
```

Comment:
if the entry point was not set, then what happens in the `_inputSignatureFunc`? Should we test this here?

Reply:
Done.